src/context/ToolboxContext/types.ts

```diff
@@ -3,25 +3,28 @@ import React from 'react';
 // eslint-disable-next-line import/no-unresolved
 import { FeatureCollection } from 'geojson';
 
-import { Collection } from '@/typings/stac';
+import { Collection, Feature as StacFeature } from '@/typings/stac';
 
 export interface ToolboxState {
   activePage: string;
   selectedCollection: Collection | null;
   selectedCollectionItems: FeatureCollection;
+  selectedCollectionItem: StacFeature;
 }
 
 export type ToolboxAction =
   | { type: 'SET_ACTIVE_PAGE'; payload: string }
   | { type: 'SET_SELECTED_COLLECTION'; payload: Collection }
-  | { type: 'SET_SELECTED_COLLECTION_ITEMS'; payload: FeatureCollection };
+  | { type: 'SET_SELECTED_COLLECTION_ITEMS'; payload: FeatureCollection }
+  | { type: 'SET_SELECTED_COLLECTION_ITEM'; payload: StacFeature };
 
 export interface ToolboxContextType {
   state: ToolboxState;
   actions: {
     setActivePage: (activePage: string) => void;
     setSelectedCollection: (selectedCollection: Collection) => void;
     setSelectedCollectionItems: (selectedCollectionItems: FeatureCollection) => void;
+    setSelectedCollectionItem: (selectedCollectionItem: StacFeature) => void;
```

Comment:
Collection item is implied, the official STAC would be just `STAC Item` and not `STAC Collection Item`, as an item is always part of a collection.

Reply:
I just kept it in line with the terminology already being used i.e `selectedCollectionItems` for all of them and `selectedCollectionItem`, which I introduced, for the currently selected one. I'd be happy for this to be changed, but I also like the correlation that the selected item is from the collection of items, not sure if that would be the same if I renamed it to `selectedStacItem`, maybe, easy enough change if everyone agreed.